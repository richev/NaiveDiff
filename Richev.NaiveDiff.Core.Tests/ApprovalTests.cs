using ApprovalTests;
using ApprovalTests.Reporters;
using Moq;
using NUnit.Framework;
using Richev.NaiveDiff.Core.Matcher;

namespace Richev.NaiveDiff.Core.Tests
{
    [UseReporter(typeof(NCrunchReporter))]
    [TestFixture]
    public class ApprovalTests
    {
        private Diff _diff;

        [SetUp]
        public void SetUp()
        {
            var matcher = new RegexMatcher(@"\d{1,2}\/\d{1,2}\/\d{4}");

            _diff = new Diff(matcher);
        }

        [Test]
        public void Foo()
        {
            var fileLeft = @"28/10/2017 Line A
28/10/2017 Line B";

            const string fileRight = @"27/10/2017  Line B
27/10/2017 Line C";

            var result = Utils.GetDiffResult(_diff, fileLeft, fileRight);

            var processed = Outputter.Process(result);

            Approvals.Verify(processed);
        }
    }
}
