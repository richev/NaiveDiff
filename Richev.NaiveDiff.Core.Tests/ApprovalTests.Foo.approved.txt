Lines.Count = 3
Lines[0].FoundIn = LeftOnly (1)
Lines[0].Line = 'Line A'
Lines[1].FoundIn = Both (0)
Lines[1].Line = 'Line B'
Lines[2].FoundIn = RightOnly (2)
Lines[2].Line = 'Line C'
